make screen output of nonlinear solver schemes more consistent

jdannberg opened the pull request:
This is a suggestion for making the screen output of the different nonlinear solver schemes more consistent. They now all output the relative nonlinear residual (which is probably more useful than the absolute residual in most cases) and the number of the current nonlinear iteration. 

The member variable `nonlinear_iteration` I added to the simulator is not really needed for this change, but I will need it for a follow-up pull request that addresses issue #300 and makes it possible to run all postprocessors also on nonlinear iterations, and it might also be useful to have the same style everywhere if we plan to mix and match solver schemes in the future.

gassmoeller commented on source/simulator/core.cc:
```diff
@@ -2153,6 +2167,13 @@ namespace aspect
+              // output relative residuals
+              pcout << "      Relative nonlinear residuals: " << relative_temperature_residual;
+              for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
+                pcout << ", " << initial_composition_residual[c];
```
This has to be `relative_composition_residual`

gassmoeller commented on source/simulator/core.cc:
```diff
@@ -2077,21 +2079,25 @@ namespace aspect
               current_linearization_point.block(introspection.block_indices.temperature)
                 = solution.block(introspection.block_indices.temperature);
               rebuild_stokes_matrix = true;
               std::vector<double> composition_residual (parameters.n_compositional_fields,0);
```
It looks like you only need the vector of relative residuals below, not the absolute residuals. So you could remove this vector and change the lines below as indicated.

gassmoeller commented on source/simulator/core.cc:
```diff
@@ -2102,11 +2108,18 @@ namespace aspect
+                        if (nonlinear_iteration == 0)
                           initial_composition_residual[c] = system_rhs.block(introspection.block_indices.compositional_fields[c]).l2_norm();
 
                         composition_residual[c]
```
you could change this into `const double composition_residual = ...`, and then use this double in line 2119.

gassmoeller commented on source/simulator/core.cc:
```diff
@@ -2153,6 +2167,13 @@ namespace aspect
 
               pcout << std::endl;
 
+              // output relative residuals
```
Why does `iterated_IMPES` also output the absolute residuals? The other schemes do not do this. Do we want to keep this behavior? Then ignore my two comments above.

jdannberg replied:
You are right, I don't really see a good reason to keep the absolute residuals for `iterated_IMPES`. If everyone is okay with that, I would just throw them out.

jdannberg commented:
I addressed your comments (and added an entry to changes.h), and I also removed the absolute residual output for iterated IMPES. 
If noone objects to that in the next few hours, I will merge it :)
